yapapi/contrib/strategy/blacklist.py

```diff
@@ -0,0 +1,75 @@
+from yapapi.strategy import MarketStrategy, SCORE_REJECTED
+from yapapi.props.builder import DemandBuilder
+from yapapi.rest.market import OfferProposal
+from typing import Awaitable, Callable, Collection, Union
+
+blacklist_type = Union[Collection[str], Callable[[], Awaitable[Collection[str]]]]
+
+
+class Blacklist(MarketStrategy):
+    """Blacklist - extend a market strategy with a provider blacklist
+
+    :param base_strategy: a market strategy that will be used to score offers from non-blacklisted providers
+    :param blacklist: either a collection of blacklisted providers ids (`offer.issuer`), or an async function that
+        returns such a collection. If a mutable collection is provided, further modifications of it will influence
+        the behaviour of the strategy.
+
+    Example 1. Add a blacklist to an a strategy::
+
+        bad_providers = ['bad_provider_1', 'bad_provider_2']
+        base_strategy = SomeStrategy(...)
+        strategy = Blacklist(base_strategy, bad_providers)
+
+    Example 2. Add a dynamic callable returning a blacklist to a strategy::
+
+        conn =  # ... create some database connection
+
+        async def get_blacklisted():
+            cursor = conn.cursor()
+            cursor.execute("SELECT provider_id FROM providers WHERE NOT allowed")
+            return [row[0] for row in cursor.fetchall()]
+
+        base_strategy = SomeStrategy()
+        strategy = Blacklist(base_strategy, get_blacklisted)
+
+
+    Example 3. Add a blacklist to the current strategy of a Golem instance::
```

Comment:
maybe we should add a specific method on Golem to perform such an operation?

Reply:
What do you mean by "such an operation"? 

1. `Golem.set_strategy()`
2. `Golem.wrap_current_strategy()`
3. `Golem.add_blacklist_to_strategy()`
?

If 1, then OK, but I don't see why such method is better than `Golem.strategy = ..` (this **will** be done via a setter function either way).
If 2, then I'm mildly against because 1 seems better. We might just want to replace current strategy with a new one, why assume we should care about the current strategy.
If 3, then I'm strongly against. There should be no `contrib`-related logic in the rest of the `yapapi` (except perhaps we might use `contrib` in defaults).

Reply:
definitely not 3 ... okay, I guess, I'm convinced by #1 ... we should just document that as a settable property...